src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java

```diff
@@ -109,12 +112,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    * Returns the shopping query by calling methods from the photo content detection classes, 
    * based on the {@code photoCategory}, passing {@code uploadedImageBytes} as argument.
    */
-  private String getQuery(String photoCategory, byte[] uploadedImageBytes) throws IllegalArgumentException {
+  private String getQuery(String photoCategory, byte[] uploadedImageBytes)
+      throws IllegalArgumentException, PhotoDetectionException { 
     switch (photoCategory) {
       case "product":
         return "Fountain pen";
       case "shopping-list":
-        return "Fuzzy socks";
+        ImageTextDectector imageTextDectector = new ImageTextDectector();
+        String shoppingQuery;
+        try {
+          shoppingQuery = imageTextDectector.imageToShoppingListExtractor(uploadedImageBytes);
+        } catch (PhotoDetectionException exception) {
+          throw exception;
+        } catch (IOException e) {
+            throw new PhotoDetectionException("Error while getting shopping query.", e);
+        }
```

Comment:
I think we can catch all IOExceptions in imageToShoppingListExtractor(..) so that only PhotoDetectionException is caught here

Reply:
I tried to catch them where ever possible in ImageTextDetector if I do more then I'd have to place try and catch blocks on every sentence in imageToShoppingListExtractor(..) (which have basically all methods).
And I think its okay to catch it here, cause I am converting it to PhotoDetectionException here too :)